test/common/router/route_fuzz_test.cc

```diff
@@ -48,22 +48,29 @@ cleanRouteConfig(envoy::config::route::v3::RouteConfiguration route_config) {
 bool validateMatcherConfig(const xds::type::matcher::v3::Matcher& matcher);
 
 bool validateOnMatchConfig(const xds::type::matcher::v3::Matcher::OnMatch& on_match) {
-  if (on_match.has_matcher() && !validateMatcherConfig(on_match.matcher())) {
-    return false;
+  switch (on_match.on_match_case()) {
+  case xds::type::matcher::v3::Matcher_OnMatch::kMatcher: {
+    return validateMatcherConfig(on_match.matcher());
   }
-  // Only envoy...Route is allowed as typed_config here.
-  if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") ==
-      std::string::npos) {
-    return false;
+  case xds::type::matcher::v3::Matcher_OnMatch::kAction: {
+    // Only envoy...Route is allowed as typed_config here.
+    if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") ==
+        std::string::npos) {
+      return false;
+    }
+    envoy::config::route::v3::Route on_match_route_action_config;
+    MessageUtil::unpackTo(on_match.action().typed_config(), on_match_route_action_config);
+    ENVOY_LOG_MISC(trace, "typed_config of on_match.action is: {}",
+                   on_match_route_action_config.DebugString());
+    return !isUnsupportedRouteConfig(on_match_route_action_config);
+  }
+  case xds::type::matcher::v3::Matcher_OnMatch::ON_MATCH_NOT_SET: {
```

Comment:
you will still need to have return statement here for this case so that all the branches all covered. Otherwise, it should generate build error

Reply:
Being in a debug build during my patching the assert(false) of course terminated the flow of control each time and was not problem. But for a release build you're right. I fixed that now. Thanks for pointing out.